vignettes/data-repository.Rmd

```diff
@@ -0,0 +1,200 @@
+data_repo <- init(path, project = "test")
+```
+
+### Storing data
+
+Use `write_delim_git()` to store a `data.frame` into the data repository. The function will separate the data and the meta data. The data is stored as a headerless, unquoted tab delimited file with ".tsv" extension and UTF-8 encoding. The meta data is stored in YAML format with ".yml" format. Therefore any extension given to the `file` will be stripped (with a warning).
+
+The function will do a lot more preprocessing to the data in order to keep the file and the diff as small as possible. That is the reason why the data is stored headerless and unquoted. 
+
+- `logical` is written as 0 (FALSE), 1 (TRUE) or NA to the data. The class is stored in the meta data.
+- `integer` is written as is to the data. The class is stored in the meta data.
+- `numeric` is written as is to the data. The class is stored in the meta data.
+- `complex` is written as is to the data. The class is stored in the meta data.
+- `character` is written as is and unquoted to the data. The class is stored in the meta data.
+- `factor` is  stored as its indices in the data. The labels of levels are stored in the meta data.
+- `POSIXct` is written as an integer to the data. The class and the origin are stored in the meta data.
+
+```{r}
+# Create dummy data
+x <- data.frame(
+    x = LETTERS,
+    y = factor(
+        sample(c("a", "b", NA), 26, replace = TRUE),
+        levels = c("a", "b", "c")
+    ),
+    z = c(NA, 1:25),
+    abc = c(rnorm(25), NA),
+    xyz = complex(real = rnorm(26), imaginary = rnorm(26)),
+    def = sample(c(TRUE, FALSE, NA), 26, replace = TRUE),
+    timestamp = seq(
+        as.POSIXct("2018-01-01"),
+        as.POSIXct("2019-01-01"),
+        length = 26
+    ),
+    stringsAsFactors = FALSE
+)
+str(x)
+```
+
+Another important part of the preprocessing is ordering both the variables as the observations. This is determined when the file is created and stored in the meta data. When the file is overwritten, the ordering is based on the existing meta data. Therefore it is important to think carefully about the ordering and make sure that is how you want it to stay **before** you create your first commit. The ordering of the observations is specified through the `sorting` argument. It holds a vector of variable names which are used in the ordering. Make sure that you add enough sorting variables to avoid ties. `sorting` also defines the order of the variables, the order of the other variables is taken from the `data.frame`.
+
+The data file (".tsv") and the meta data file (".yml") are automatically staged after writing. `write_delim_git()` returns the hashes of both the data file and the meta data file. The names of the vector contain the path of the files, relative to the root of the _git_ repository.
+
+```{r}
+status(data_repo)
+write_delim_git(x = x, file = "my_data", repo = data_repo, sorting = c("y", "x"))
+status(data_repo)
+status(repo)
```

Comment:
As a user, I would rather use only `status(repo)` and get the information (in the background a difference can be made to prepare the output, but my output as user should just require the general repo-concept)